AssemblerLib/MeshComponentMap.cpp

```diff
@@ -55,12 +69,23 @@ MeshComponentMap::MeshComponentMap(
                 // mesh items are ordered first by node, cell, ....
                 for (std::size_t j=0; j<mesh_subset.getNNodes(); j++)
                 {
-                    const GlobalIndexType global_id = static_cast<GlobalIndexType>(components.size()
+                    GlobalIndexType global_id = static_cast<GlobalIndexType>(components.size()
```

Comment:
This was not introduced in this PR, but the formula for `global_id` looks like it makes some assumption about the component order. Btw: The argument `order` seems to be unused.